added new Pin Headers

Shackmeister opened the pull request:
![pin_headers](https://user-images.githubusercontent.com/5279723/33515720-3a42af74-d767-11e7-874b-16c90a248147.png)

poeschlr commented:
What did you use for the license string?

I think i will use the following:

```
LIST_license = ["Copyright (C) "+datetime.now().strftime("%Y")+", " + STR_licAuthor,
                "",
                "The KiCad libraries are licensed under the Creative Commons CC-BY-SA 4.0 License,",
                "with the following exception:",
                "To the extent that the creation of electronic designs that use 'Licensed Material'",
                "can be considered to be 'Adapted Material', then the copyright holder waives article 3",
                "of the license with respect to these designs and any generated files which use data", "provided as part of the 'Licensed Material'.",
                ""
                ]
```

Shackmeister commented:
I use the one @maui made for the scripts :)
https://github.com/easyw/kicad-3d-models-in-freecad/blob/9814e0b63f05c32665da5ffac43f3b0b4a09cbc9/cadquery/FCAD_script_generator/_tools/add_license.py

poeschlr commented:
Thanks. (I did not notice that maui already fixed it. To be honest i am not even sure if my script uses the license text within the script or the central text.)

Shackmeister commented:
Fair enough :) if you don't have any comments on hus license text it would probably be better to streamline it in the repo :)
BTW a new yaml Resistors_SMD script has been made for the footprints, Ill make a PR within a few days

poeschlr commented:
I already streamlined nearly all footprint scripts. I fear i don't have the time to streamline the 3d model scripts as well. (I make as little changes as possible just to ensure that the models are created correctly.)

Shackmeister commented:
@poeschlr, maui has just finished a script which checks the geometry when creating the models :)

SchrodingersGat commented:
@easyw can this script be run external to FreeCAD?

easyw commented:
@SchrodingersGat 
the BOP checker can run only inside FreeCAD... I'm not aware if that can be run even without a FreeCADGui ... before recent release of FC the checking would not was available with python.

I'm thinking that we could add a template to the PR in which the user would have some check list to mark...
FreeCAD has exactly this kind of check list to help people in presenting a well formed PR
![pr-checklist](https://user-images.githubusercontent.com/3032347/33530471-3f626958-d880-11e7-83ee-30e9cddc7064.jpg)
 
We could put some checking like:
- [ ] is your 3D model made in a mechanical modeler?
- [ ] have you submitted the source of the model (source file or script)?
- [ ] have you submitted both STEP in scale 1:1 and wrl in scale 1:2.54?
- [ ] are your 3D models aligned to the KiCad footprint?
- [ ] is your WRL model using material properties?
- [ ] did you run a geometry checking on the mechanical model?

(we could give instructions for the last point in the README of the repo....)

Shackmeister commented:
Hi guys I just added the new check to the script, but unfortunately my script doesnt run in 0.17 before the next daily build comes out..
I would like to test the script out, so please dont merge yet :)

But please feel free to check the models with regards to the parameters
test board here: 
[Pin_Header_Test.kicad_pcb.txt](https://github.com/KiCad/kicad-packages3D/files/1525249/Pin_Header_Test.kicad_pcb.txt)

Shackmeister commented:
this is ready for review and merge as soon as my computer finishes pushing it

poeschlr commented:
I think i have bad news for you. The library name is incorrect. Should be Connector_* not Conn_*

Shackmeister commented:
Alrighty, ill update this and get back

Shackmeister commented:
@poeschlr 
I renamed the folders, and fixed the testboard

[Pin_Header_Test.kicad_pcb.txt](https://github.com/KiCad/kicad-packages3D/files/1553620/Pin_Header_Test.kicad_pcb.txt)

poeschlr commented:
Looks ok now.
By the way i moved the geometry check management stuff as a separate function over to the cq_cad_tools fie. This function now also includes some better error handling. More details see: https://github.com/KiCad/kicad-packages3D/pull/201
